integrations/neo4j-document-store.md

```diff
@@ -0,0 +1,225 @@
+
+The `neo4j-haystack` library uses [Python Driver](https://neo4j.com/docs/api/python-driver/current/api.html#api-documentation) and
+[Cypher Queries](https://neo4j.com/docs/cypher-manual/current/introduction/) to interact with Neo4j database and hide all complexities under the hood.
+
+`Neo4jDocumentStore` will store Documents as Graph nodes in Neo4j. Embeddings are stored as part of the node, but indexing and querying of vector embeddings using ANN is managed by a dedicated [Vector Index](https://neo4j.com/docs/cypher-manual/current/indexes-for-vector-search/).
+
+```text
+                                   +-----------------------------+
+                                   |       Neo4j Database        |
+                                   +-----------------------------+
+                                   |                             |
+                                   |      +----------------+     |
+                                   |      |    Document    |     |
+                write_documents    |      +----------------+     |
+          +------------------------+----->|   properties   |     |
+          |                        |      |                |     |
++---------+----------+             |      |   embedding    |     |
+|                    |             |      +--------+-------+     |
+| Neo4jDocumentStore |             |               |             |
+|                    |             |               |index/query  |
++---------+----------+             |               |             |
+          |                        |      +--------+--------+    |
+          |                        |      |  Vector Index   |    |
+          +----------------------->|      |                 |    |
+               query_embeddings    |      | (for embedding) |    |
+                                   |      +-----------------+    |
+                                   |                             |
+                                   +-----------------------------+
+```
+
+In the above diagram:
+
+- `Document` is a Neo4j node (with "Document" label)
+- `properties` are Document [attributes](https://docs.haystack.deepset.ai/v2.0/docs/data-classes#document) stored as part of the node.
+- `embedding` is also a property of the Document node (just shown separately in the diagram for clarity) which is a vector of type `LIST[FLOAT]`.
+- `Vector Index` is where embeddings are getting indexed by Neo4j as soon as those are updated in Document nodes.
+
+## Installation
+
+`neo4j-haystack` can be installed as any other Python library, using pip:
+
+```bash
+pip install --upgrade pip # optional
+pip install neo4j-haystack
+```
```

Comment:
I think there is a dependency on `sentence-transformers` as well in order to run the code samples below?